code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm

```diff
@@ -64,54 +68,71 @@
 		return TRUE
 
 	if(isobj(the_target) && is_type_in_typecache(the_target, wanted_objects))
-		return TRUE
+		if(udder.reagents.total_volume < udder.reagents.maximum_volume)
```

Comment:
you should use signals for something like this instead to replace the need for a reference.

COMSIG_IS_UDDER_FULL or something